<launch>
  <arg name="robot_name" default="trilosaurus" />
  
  <arg name="start_hardware" default="true" />
  <arg name="start_camera" default="true" />
  <arg name="start_lane_detector" default="true" />
  <arg name="start_lane_guidance" default="true" />
  <arg name="start_calibration" default="false" />
  <arg name="start_pure_pursuit_guidance" default="true" />

  <!-- ****** Starts Hardware ****** -->
  <group if="$(arg start_hardware)">
    <node name="hardware_node" pkg="trilosaurus_bringup" type="robot_manager_ros.py"/>
  </group>
  
  <!-- ****** Starts Camera ****** -->
  <group if="$(arg start_camera)">
    <arg name="ref_to_camo_t" default="0 0 0"/>
    <arg name="ref_to_camo_q" default="1 0 0 0"/>

    <include file="$(find trilosaurus_bringup)/launch/camera.launch">
      <arg name="enable_raw" value="true"/>
    </include>

    </group>

    <!-- ****** Starts Lane Detector ****** -->
    <group if="$(arg start_lane_detector)">
      <node name="lane_detector" pkg="common_vision" type="lane_detector_node2.py"  output="screen">
	<param name="camera" value="camera1"/>
	<param name="robot_name" value="trilopi"/>
	<param name="ref_frame" value="base_link_footprint"/>
      </node>
    </group>

    <!-- ****** Starts Lane Guidance ****** -->
    <group if="$(arg start_lane_guidance)">
      <node name="lane_guidance" pkg="trilosaurus_bringup" type="line_guidance.py">
     	<param name="camera" value="camera1"/>
	<param name="robot_name" value="trilopi"/>
	<param name="ref_frame" value="base_link_footprint"/>
	<param name="lookahead_dist" value="0.13"/>
	<param name="vel_sp" value="0.03"/>
      </node>
    </group>
    
    <!-- ****** Starts Calibration ****** -->
    <group if="$(arg start_calibration)">
      <include file="$(find trilosaurus_bringup)/launch/calibration.launch">
      </include>
    </group>

    <!-- ****** Starts PurePursuit Guidance ****** -->
    <group if="$(arg start_pure_pursuit_guidance)">
      <include file="$(find trilosaurus_bringup)/launch/pure_pursuit_guidance.launch">
      </include>
    </group>



    
    
</launch>


